bitcoin/src/p2p/mod.rs

```diff
@@ -242,39 +244,31 @@ impl FromStr for Magic {
         }
     }
 }
-
-macro_rules! generate_network_magic_conversion {
-    ($(Network::$network:ident => Magic::$magic:ident,)*) => {
-        impl From<Network> for Magic {
-            fn from(network: Network) -> Magic {
-                match network {
-                    $(
-                        Network::$network => Magic::$magic,
-                    )*
-                }
-            }
+impl From<Network> for Magic {
+    fn from(network: Network) -> Magic {
+        match network {
+            Network::Bitcoin => Magic::BITCOIN,
+            Network::Testnet(TestnetVersion::V3) => Magic::TESTNET,
+            Network::Testnet(TestnetVersion::V4) => Magic::TESTNET4,
+            Network::Signet => Magic::SIGNET,
+            Network::Regtest => Magic::REGTEST,
         }
+    }
+}
 
-        impl TryFrom<Magic> for Network {
-            type Error = UnknownMagicError;
+impl TryFrom<Magic> for Network {
+    type Error = UnknownMagicError;
 
-            fn try_from(magic: Magic) -> Result<Self, Self::Error> {
-                match magic {
-                    $(
-                        Magic::$magic => Ok(Network::$network),
-                    )*
-                    _ => Err(UnknownMagicError(magic)),
-                }
-            }
+    fn try_from(magic: Magic) -> Result<Self, Self::Error> {
+        match magic {
+            Magic::BITCOIN => Ok(Network::Bitcoin),
+            Magic::TESTNET => Ok(Network::Testnet(TestnetVersion::V3)),
+            Magic::TESTNET4 => Ok(Network::Testnet(TestnetVersion::V4)),
+            Magic::SIGNET => Ok(Network::Signet),
+            Magic::REGTEST => Ok(Network::Regtest),
+            _ => Err(UnknownMagicError(magic)),
         }
-    };
-}
-
-generate_network_magic_conversion! {
-    Network::Bitcoin => Magic::BITCOIN,
-    Network::Testnet => Magic::TESTNET,
-    Network::Signet => Magic::SIGNET,
-    Network::Regtest => Magic::REGTEST,
+    }
```

Comment:
Losing this sucks as it prevented us from creating bugs. It'd be great to somehow restore the macro.

Reply:
Ok, I will try to reproduce this macro and try to split it into two macros, because the 'Network' variant we matched cannot satisfy the two internal logics at the same time